hoodie-client/src/test/java/com/uber/hoodie/common/HoodieClientTestUtils.java

```diff
@@ -87,7 +87,7 @@ public static SparkConf getSparkConfForTest(String appName) {
         SparkConf sparkConf = new SparkConf()
                 .setAppName(appName)
                 .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
-                .setMaster("local[4]");
+                .setMaster("local[1]");
```

Comment:
are you trying to cut down parallelism to avoid race.. if so, it just seems to be work around the problem.. a real use can set cores > 1 and this could happen?